src/SFA.DAS.Assessor.Functions.ExternalApis/DataCollection/Authentication/DataCollectionTokenService.cs

```diff
@@ -1,29 +1,42 @@
 ﻿using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
+using System;
 
 namespace SFA.DAS.Assessor.Functions.ExternalApis.DataCollection.Authentication
 {
     public class DataCollectionTokenService : IDataCollectionTokenService
     {
-        private readonly IDataCollectionServiceAnonymousApiClient _dataCollectionServiceAnonymousApiClient;
         private readonly IOptions<DataCollectionApiAuthentication> _dataCollectionApiAuthenticationOptions;
-        
-        public DataCollectionTokenService(IDataCollectionServiceAnonymousApiClient dataCollectionServiceAnonymousApiClient, IOptions<DataCollectionApiAuthentication> dataCollectionApiAuthenticationOptions)
+                
+        public DataCollectionTokenService(IOptions<DataCollectionApiAuthentication> dataCollectionApiAuthenticationOptions)
         {
-            _dataCollectionServiceAnonymousApiClient = dataCollectionServiceAnonymousApiClient;
-            _dataCollectionApiAuthenticationOptions = dataCollectionApiAuthenticationOptions;
+            _dataCollectionApiAuthenticationOptions = dataCollectionApiAuthenticationOptions;    
         }
 
         public string GetToken()
         {
-            var tokenRequest = new DataCollectionTokenRequest
-            {
-                Username = _dataCollectionApiAuthenticationOptions?.Value.Username,
-                Password = _dataCollectionApiAuthenticationOptions?.Value.Password
-            };
+            var clientId = _dataCollectionApiAuthenticationOptions?.Value.ClientId;
+            var clientSecret = _dataCollectionApiAuthenticationOptions?.Value.ClientSecret;
+            var scope = _dataCollectionApiAuthenticationOptions?.Value.Scope;
+
+            IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(clientId)
+                    .WithClientSecret(clientSecret)
+                    .WithAuthority(new Uri(_dataCollectionApiAuthenticationOptions?.Value.Authority))
+                    .Build();
+
+            string[] scopes = new string[] { scope };
 
-            var t = _dataCollectionServiceAnonymousApiClient.GetToken(tokenRequest);
+            AuthenticationResult result;
+            try
+            {
+                result = app.AcquireTokenForClient(scopes).ExecuteAsync().Result;
```

Comment:
Should we await this async call and make this method async, returning a Task ?

Reply:
This has now been Async using a DelegatingHandler from the HttpClient pattern